src/openforms/registrations/contrib/email/config.py

```diff
@@ -91,3 +94,30 @@ class Meta:
                 "email_content_template_text",
             ),
         ]
+
+
+class Options(TypedDict):
+    """
+    Shape of the email registration plugin options.
+
+    This describes the shape of :attr:`EmailOptionsSerializer.validated_data`, after
+    the input data has been cleaned/validated.
+    """
+
+    to_emails: list[str]
+    attachment_formats: NotRequired[list[AttachmentFormat | str]]
+    payment_emails: NotRequired[list[str]]
+    attach_files_to_email: bool | None
+    email_subject: NotRequired[str]
+    email_payment_subject: NotRequired[str]
+    email_content_template_html: NotRequired[str]
+    email_content_template_text: NotRequired[str]
+
+
+# sanity check for development - keep serializer and type definitions in sync
+_serializer_fields = EmailOptionsSerializer._declared_fields.keys()
+_options_keys = Options.__annotations__.keys()
+assert (
+    _serializer_fields == _options_keys
+), "Mismatch between serializer fields and options dictionary keys!"
+del _serializer_fields, _options_keys
```

Comment:
Interesting :thinking:

Reply:
it's disgusting and it will slightly affect the application startup, but it's a hit worth taking :grimacing:

Reply:
Even hackier, this check could be moved to the plugin class, by using the `serialization_options` class var and the parametrized options class

Reply:
good shout, though we can only do that once all plugins are updated with the options annotation :thinking: I'll keep it in mind for when we get there!